run_linux.go

```diff
@@ -557,19 +557,52 @@ func (b *Builder) setupMounts(mountPoint string, spec *specs.Spec, bundlePath st
 	return runTargets, nil
 }
 
-// addNetworkConfig copies files from host and sets them up to bind mount into container
-func (b *Builder) addNetworkConfig(rdir, hostPath string, chownOpts *idtools.IDPair, dnsServers, dnsSearch, dnsOptions []string, namespaceOptions define.NamespaceOptions) (string, error) {
-	stat, err := os.Stat(hostPath)
+// addResolvConf copies files from host and sets them up to bind mount into container
+func (b *Builder) addResolvConf(rdir string, chownOpts *idtools.IDPair, dnsServers, dnsSearch, dnsOptions []string, namespaceOptions define.NamespaceOptions) (string, error) {
+	resolvConf := "/etc/resolv.conf"
+
+	stat, err := os.Stat(resolvConf)
 	if err != nil {
 		return "", err
 	}
-	contents, err := ioutil.ReadFile(hostPath)
-	if err != nil {
+	contents, err := ioutil.ReadFile(resolvConf)
+	// resolv.conf doesn't have to exists
+	if err != nil && !os.IsNotExist(err) {
 		return "", err
 	}
 
+	netns := false
+	ns := namespaceOptions.Find(string(spec.NetworkNamespace))
+	if ns != nil && !ns.Host {
+		netns = true
+	}
+
+	nameservers := resolvconf.GetNameservers(contents, types.IPv4)
+	// check if systemd-resolved is used, assume it is used when 127.0.0.53 is the only nameserver
+	if len(nameservers) == 1 && nameservers[0] == "127.0.0.53" && netns {
```

Comment:
Is there a test for when only this nameserver is up?  I don't see one below, but it's late.

Reply:
IDK, but can the user change the IP for the system nameserver?  It might be nice to have this value stored in a config file that could be tweaked.  I'm also a little twitchy about having a particular IP specified and having someone spoof it, but I can't see the harm atm in that case.

Reply:
No this is correct like this. It now does the same as podman. The problem with systemd-resolved is that it only sets `127.0.0.53` as nameserver in `/etc/resolv.conf`, this is a hard coded value AFAICT and there is no reason why someone should change this. The actual nameservers are in `/run/systemd/resolve/resolv.conf`. We have to read the actual nameservers in this case and filter the 127... addresses out because there are not reachable in a new netns. If there is no netns we can use the system one because 127.0.0.53 is reachable.
Fedora uses systemd-resolved so the CI should test this.